compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -976,6 +976,9 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
         stack: &TraitObligationStack<'o, 'tcx>,
         candidate: &SelectionCandidate<'tcx>,
     ) -> Result<EvaluationResult, OverflowError> {
+        if matches!(candidate, SelectionCandidate::LazyCandidate) {
```

Comment:
I think you should not select any candidates in assemble_candidates_from_unified_impls, and instead of bailing out successfully here, actually go into confirmation. Confirmation can then generate obligations for all the variants of the enum/bool

Reply:
So what you're saying is instead of 1 single fake candidates, generate a new obligation that looks for all candidates of the variants?

Reply:
I think you'll still want one candidate, so that we first look for a single generic impl with the existing candidates, and if that fails, we process your new candidate in confirmation by spewing an obligation per variant